screens/recipes/RecipesScreen.js

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable max-statements */
+
+import React, { useEffect, useState } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { useStoreActions } from 'easy-peasy';
+import { Icon } from 'react-native-elements';
+import colors from '../../styles/appColors';
+import styles from '../../styles/Recipes/index';
+
+function Recipes(props) {
+  const { navigation } = props;
+  const getRecipes = useStoreActions((actions) => actions.getRecipes);
+  const [recipes, setRecipes] = useState([]);
+  const [showError, setShowError] = useState(false);
+  const [newRecipe, setNewRecipe] = useState(false);
+  const [rows, setRows] = useState([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    getRecipes()
+      .then((res) => {
+        setRecipes(res);
+      })
+      .catch((err) => {
+        setShowError(true);
+        setErrorMessage(err);
+      });
+  }, [newRecipe]);
+
+  useEffect(() => {
+    const auxRows = [];
+    for (let i = 0;
+      i < recipes.length; // - totalRows < 0) ? i < totalRows : i < recipes.length;
+      i++) {
+      auxRows.push(
+        <TouchableOpacity
+          style={styles.recipeRow}
+          onPress={() => navigation.navigate('Receta', recipes[i].attributes)}
+          key={recipes[i].id}>
+          <View style={styles.left}>
+            <Text style={styles.name} >{(recipes[i]) ? recipes[i].attributes.name : '---'}</Text>
+            <View style={styles.recipeInfo}>
+              <Icon name='pie-chart' color={colors.recipeIcon} size='23' />
+              <Text style = {styles.subtitle}>
+                {recipes[i].attributes.portions} {(recipes[i].attributes.portions === 1 ? 'porción' : 'porciones')}
+              </Text>
+            </View>
+            <View style={styles.recipeInfo}>
+              <Icon name='timer' color={colors.recipeIcon} size='23' />
+              <Text style = {styles.subtitle}>{recipes[i].attributes.cook_minutes} minutos</Text>
+            </View>
+          </View>
+          <View style={styles.right}>
+            <Text style = {styles.price}>$XX.XXX</Text>
+          </View>
+        </TouchableOpacity>,
+      );
+    }
+    setRows(auxRows);
+  }, [recipes]);
+
+  if (recipes.length) {
+    return (
+      <>
+        {rows}
```

Comment:
Creo que no hay necesidad de la variable secundaria rows, podrias hacer acá mismo

```js
if (recipes.length) {
  return recipes.map((recipe) => (
    <RecipeComponent recipe={recipe} />
  )
}
``` 

Algo así

Reply:
Esto quiere decir que en vez de hacer un for, creo una componente que sea solo un TouchableOpacity y luego realizo un map sobre todo esta lista creando por cada "objeto" uno de estos Touchable?

Reply:
Done